### PR TITLE
Multiple improvements to the Metrics SDK

### PR DESCRIPTION
**Pull request by reyang**
## Changes

* Added concurrency/compatibilty requirements section in the Metrics SDK
* Added `MetricReader.Shutdown`
* Updated the ToC

### CODE REVIEW

**Review by hdost — APPROVED** [review]

File: specification/metrics/sdk.md [hunk 1]
```diff
@@ -766,3 +784,28 @@ Known values for `OTEL_METRICS_EXEMPLAR_FILTER` are:
...
 - `"ALL"`: All measurements are eligible for exemplar sampling.
 - `"WITH_SAMPLED_TRACE"`: Only allow measurements with a sampled parent span in context.
+
+## Compatibility requirements
+
+All the metrics components SHOULD allow new methods to be added to existing
+components without introducing breaking changes.
+
+All the metrics SDK methods SHOULD allow optional parameter(s) to be added to
+existing methods without introducing breaking changes.
```
Comment (pirgeo) [inline specification/metrics/sdk.md:794]: Is there a way to enforce this in languages like Go where there are no optional parameters or function overloading?

Comment (jmacd) [reply specification/metrics/sdk.md:794]:
```suggestion
All the metrics SDK methods SHOULD allow optional parameter(s) to be added to
existing methods without introducing breaking changes, if possible.
```

**Review by jmacd — APPROVED** [review]

**Review by pirgeo — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Review by bogdandrutu — COMMENTED** [review]
FYI: This is no longer "minor improvement", it is actually fixing some missing definitions and adds content for concurrency.

**Comment by jmacd** [comment]
Looks good -- will you resolve the conflicts?

**Review by cijothomas — APPROVED** [review]

**Comment by carlosalberto** [comment]
LGTM - reconsider adding the concurrency change in the CHANGELOG entry, btw.

**Comment by reyang** [comment]
> LGTM - reconsider adding the concurrency change in the CHANGELOG entry, btw.

Done.

**Pull request merged**